### ISSUE TITLE
APIError(code=-1100): Illegal characters found in parameter 'price'

### ISSUE THREAD

**Issue by ReinisL** [ccxt/binance-trade-bot#228]
This error appears when Bot tries to 
`crypto_trading_logger - INFO - Sold DASH`
`crypto_trading_logger - INFO - BUY QTY 37775.0`

Here is the API error which goes in to loop 

`APIError(code=-1100): Illegal characters found in parameter 'price'; legal range is '^([0-9]{1,20})(\.[0-9]{1,20})?$'.`

**Comment by Pekafu** [comment]
Do you have more rows from the log? This seems it has managed to sell those DASH(es) however it's trying to buy a lot of something which probably has really small price. perhaps bot is getting the current price from the binance with x*10^-y and passing this back to binance api causes issues. We might need to add some sort of converter in case

**Comment by ReinisL** [comment]
@Pekafu Yes, here's the full log of this case

`2021-03-22 21:27:04,878 - crypto_trading_logger - INFO - Starting`
`2021-03-22 21:27:05,445 - crypto_trading_logger - INFO - Creating database schema if it doesn't already exist`
`2021-03-22 21:27:06,173 - crypto_trading_logger - INFO - Setting initial coin to DASH`
`2021-03-22 21:27:06,189 - crypto_trading_logger - INFO - Purchasing <DASH> to begin trading`
`2021-03-22 21:27:09,208 - crypto_trading_logger - INFO - BUY QTY 0.573`
`2021-03-22 21:27:09,751 - crypto_trading_logger - INFO - {'symbol': 'DASHBNB', 'orderId': 60973490, 'orderListId': -1, 'clientOrderId': 'eTUlBdXqyW0nS8gwDESCiT', 'transactTime': 1616441229534, 'price': '0.81400000', 'origQty': '0.57300000', 'executedQty': '0.00000000', 'cummulativeQuoteQty': '0.00000000', 'status': 'NEW', 'timeInForce': 'GTC', 'type': 'LIMIT', 'side': 'BUY', 'fills': []}`
`2021-03-22 21:27:10,233 - crypto_trading_logger - INFO - APIError(code=-2013): Order does not exist.`
`2021-03-22 21:27:11,537 - crypto_trading_logger - INFO - {'symbol': 'DASHBNB', 'orderId': 60973490, 'orderListId': -1, 'clientOrderId': 'eTUlBdXqyW0nS8gwDESCiT', 'price': '0.81400000', 'origQty': '0.57300000', 'executedQty': '0.57300000', 'cummulativeQuoteQty': '0.46642200', 'status': 'FILLED', 'timeInForce': 'GTC', 'type': 'LIMIT', 'side': 'BUY', 'stopPrice': '0.00000000', 'icebergQty': '0.00000000', 'time': 1616441229534, 'updateTime': 1616441229543, 'isWorking': True, 'origQuoteOrderQty': '0.00000000'}`
`2021-03-22 21:27:11,538 - crypto_trading_logger - INFO - Bought DASH`
`2021-03-22 21:27:11,550 - crypto_trading_logger - INFO - Ready to start trading`
`2021-03-22 21:28:13,385 - crypto_trading_logger - INFO - Will be jumping from <DASH> to BTT`
`2021-03-22 21:28:14,941 - crypto_trading_logger - INFO - Selling 0.572 of DASH`
`2021-03-22 21:28:14,941 - crypto_trading_logger - INFO - Balance is 0.572427`
`2021-03-22 21:28:15,213 - crypto_trading_logger - INFO - order`
`2021-03-22 21:28:15,214 - crypto_trading_logger - INFO - {'symbol': 'DASHBNB', 'orderId': 60973612, 'orderListId': -1, 'clientOrderId': 'LUvcg6d0kPV6oCGdN5ZLiK', 'transactTime': 1616441295275, 'price': '0.00000000', 'origQty': '0.57200000', 'executedQty': '0.57200000', 'cummulativeQuoteQty': '0.46301500', 'status': 'FILLED', 'timeInForce': 'GTC', 'type': 'MARKET', 'side': 'SELL', 'fills': [{'price': '0.81000000', 'qty': '0.26700000', 'commission': '0.00016220', 'commissionAsset': 'BNB', 'tradeId': 744684}, {'price': '0.80900000', 'qty': '0.30500000', 'commission': '0.00018506', 'commissionAsset': 'BNB', 'tradeId': 744685}]}`
`2021-03-22 21:28:15,226 - crypto_trading_logger - INFO - Waiting for Binance`
`2021-03-22 21:28:15,489 - crypto_trading_logger - INFO - APIError(code=-2013): Order does not exist.`
`2021-03-22 21:28:16,752 - crypto_trading_logger - INFO - {'symbol': 'DASHBNB', 'orderId': 60973612, 'orderListId': -1, 'clientOrderId': 'LUvcg6d0kPV6oCGdN5ZLiK', 'price': '0.00000000', 'origQty': '0.57200000', 'executedQty': '0.57200000', 'cummulativeQuoteQty': '0.46301500', 'status': 'FILLED', 'timeInForce': 'GTC', 'type': 'MARKET', 'side': 'SELL', 'stopPrice': '0.00000000', 'icebergQty': '0.00000000', 'time': 1616441295275, 'updateTime': 1616441295275, 'isWorking': True, 'origQuoteOrderQty': '0.00000000'}`
`2021-03-22 21:28:17,019 - crypto_trading_logger - INFO - Sold DASH`
`2021-03-22 21:28:18,583 - crypto_trading_logger - INFO - BUY QTY 37775.0`
`2021-03-22 21:28:18,850 - crypto_trading_logger - INFO - APIError(code=-1100): Illegal characters found in parameter 'price'; legal range is '^([0-9]{1,20})(\.[0-9]{1,20})?$'.`

**Comment by Pekafu** [comment]
yeah so basically BTTBNB is 0.00001177 which could be converted into a 1.177*10^-5, this binance is giving this value to us but not allowing us to reply with same format. So converter is required on our side

**Comment by jtwebguy** [comment]
Hi, any update on this?

**Comment by jtwebguy** [comment]
I'm not a phyton guy but i found a solution:

https://github.com/sammchardy/python-binance/issues/212

Hope this gets resolved. thanks

**Comment by teatoroomtwo** [comment]
Hello guys, any news related to this problem?
How can we avoid this beforehand until the bug is resolved?

**Comment by habibalkhabbaz** [comment]
#276 fixing this issue

**Comment by Azharthegreat** [comment]
I have still same issue, please help

**Comment by rroblik** [comment]
up !
